lib/internal/util/inspect.js

```diff
@@ -861,8 +862,32 @@ function formatPrimitive(fn, value, ctx) {
   return fn(value.toString(), 'symbol');
 }
 
-function formatError(value) {
-  return value.stack || errorToString(value);
+function formatError(value, constructorName) {
+
+  if (value.stack) {
+    let stack = value.stack;
+    if (!stack.startsWith(constructorName) &&
+        !(stack.startsWith('file:') && !constructorName.startsWith('file'))) {
+      const stackStart = value.stack.indexOf('\n    at');
+      const errorAndMessage = value.stack.slice(0, stackStart);
+      const messageStart = errorAndMessage.indexOf(':');
+      if (stack && stackStart !== -1) {
```

Comment:
Do we need to check for `stack` here? Seems like it's already covered on [L867](https://github.com/nodejs/node/pull/23852/files#diff-91f6ca3fae2551d5959872b74c18946aR867)

```suggestion
      if (stackStart !== -1) {
```

Reply:
Yes, you are right. That's not needed, will change it.

Reply:
Good catch, I wasn't aware of this. Yes we need to handle `Error.stackTraceLimit` to 0 cases as well